pysph/examples/gas_dynamics/shocktube.py

```diff
@@ -55,42 +76,169 @@ def create_particles(self):
+        rho = numpy.ones_like(x) * self.rhol
+        rho[right_indices] = self.rhor
 
         # pl = 100.0, pr = 0.1
-        p = numpy.ones_like(x) * 1000.0
-        p[right_indices] = 0.01
+        p = numpy.ones_like(x) * self.pl
+        p[right_indices] = self.pr
 
         # const h and mass
-        h = numpy.ones_like(x) * h0
+        h = numpy.ones_like(x) * self.hdx * self.dx
         m = numpy.ones_like(x) * volume * rho
 
+        # ul = ur = 0
+        u = numpy.ones_like(x) * self.ul
+        u[right_indices] = self.ur
+
+        # vl = vr = 0
+        v = numpy.ones_like(x) * self.vl
+        v[right_indices] = self.vr
+
         # thermal energy from the ideal gas EOS
         e = p/(gamma1*rho)
 
         fluid = gpa(name='fluid', x=x, y=y, rho=rho, p=p, e=e, h=h, m=m,
-                    h0=h.copy())
+                    h0=h.copy(), u=u, v=v)
         self.scheme.setup_properties([fluid])
 
         print("2D Shocktube with %d particles"%(fluid.get_number_of_particles()))
 
         return [fluid,]
 
     def create_scheme(self):
-        s = GasDScheme(
+        self.dt = dt
+        self.tf = tf
+        
+        adke = ADKEScheme(
+            fluids=['fluid'], solids=[], dim=dim, gamma=gamma,
+            alpha=1, beta=1, k=1.0, eps=0.8, g1=0.5, g2=0.5)
+
+        mpm = GasDScheme(
             fluids=['fluid'], solids=[], dim=dim, gamma=gamma,
             kernel_factor=kernel_factor, alpha1=alpha1, alpha2=alpha2,
             beta=beta
         )
-        s.configure_solver(dt=dt, tf=tf, adaptive_timestep=False, pfreq=50)
+
+        gsph = GSPHScheme(
+            fluids=['fluid'], solids=[], dim=dim, gamma=gamma,
+            kernel_factor=1.5,
+            g1=0.25, g2=0.5, rsolver=2, interpolation=1, monotonicity=2,
+            interface_zero=True, hybrid=False, blend_alpha=2.0,
+            niter=40, tol=1e-6
+        )
+
+        s = SchemeChooser(
+            default='adke', adke=adke, mpm=mpm, gsph=gsph
+        )
         return s
+    
+    def configure_scheme(self):
+        s = self.scheme
+        if self.options.scheme == 'mpm':
+            s.configure(kernel_factor=kernel_factor)
+            s.configure_solver(dt=self.dt, tf=self.tf,
+                               adaptive_timestep=True, pfreq=50)
+        elif self.options.scheme == 'adke':
+            s.configure_solver(dt=self.dt, tf=self.tf,
+                               adaptive_timestep=False, pfreq=50)
+        elif self.options.scheme == 'gsph':
+            s.configure_solver(dt=self.dt, tf=self.tf,
+                               adaptive_timestep=False, pfreq=50)
+
+    def post_process(self):
+        try:
+            from matplotlib import pyplot
+        except ImportError:
+            print("Post processing requires matplotlib.")
+            return
+        import os
+        from pysph.solver.utils import load
+        from pysph.examples.gas_dynamics import riemann_solver
+        outfile = self.output_files[-1]
```

Comment:
You need to guard for this, as the tests do not generate any output, so if `len(self.output_files)` is 0 then you should just return.